src/EnergyPlus/EconomicLifeCycleCost.cc

```diff
@@ -1149,28 +1140,26 @@ void ExpressAsCashFlows(EnergyPlusData &state)
 
     // gather costs from EconomicTariff for each end use
     elcc->numResourcesUsed = 0;
-    for (ResourceType iResource : state.dataGlobalConst->AllResourceTypes) {
-        GetMonthlyCostForResource(state, iResource, curResourceCosts);
+    for (int iResource = 0; iResource < static_cast<int>(Constant::ResourceType::Num); ++iResource) { 
+        GetMonthlyCostForResource(state, static_cast<Constant::ResourceType>(iResource), curResourceCosts);
         annualCost = 0.0;
         for (int jMonth = 1; jMonth <= 12; ++jMonth) {
-            resourceCosts.at(jMonth).at(iResource) = curResourceCosts(jMonth);
-            annualCost += resourceCosts.at(jMonth).at(iResource);
+            resourceCosts[jMonth][iResource] = curResourceCosts(jMonth);
+            annualCost += resourceCosts[jMonth][iResource];
         }
         if (annualCost != 0.0) {
             ++elcc->numResourcesUsed;
-            resourceCostNotZero.at(iResource) = true;
+            resourceCostNotZero[iResource] = true;
         } else {
-            resourceCostNotZero.at(iResource) = false;
+            resourceCostNotZero[iResource] = false;
         }
-        resourceCostAnnual.at(iResource) = annualCost;
+        resourceCostAnnual[iResource] = annualCost;
     }
     // allocate the escalated energy cost arrays
     for (int year = 1; year <= elcc->lengthStudyYears; ++year) {
-        std::map<DataGlobalConstants::ResourceType, Real64> yearMap;
-        for (ResourceType iResource : state.dataGlobalConst->AllResourceTypes) {
-            yearMap.insert(std::pair<DataGlobalConstants::ResourceType, Real64>(iResource, 0.0));
-        }
-        elcc->EscalatedEnergy.insert(std::pair<int, std::map<DataGlobalConstants::ResourceType, Real64>>(year, yearMap));
+        std::array<Real64, static_cast<int>(Constant::ResourceType::Num)> yearMap;
+        std::fill(yearMap.begin(), yearMap.end(), 0.0);
+        elcc->EscalatedEnergy[year] = yearMap;
     }
```

Comment:
Don't need to create the yearMap array 12 times, only the EscalatedEnergy array needs to be filled with that array 12 times.

Reply:
You do need to create it 12 times. Otherwise, all entries in EscalatedEnergy will point to the same object and overwrite one another.